### ISSUE TITLE
Example of Axage with ABAP2UI5

### ISSUE THREAD

**Issue by jung-thomas** [Ennowulff/axage#6]
Using the new version in the Pull Request (that has ABAP Cloud Class based approach and serializable instances), I've created a stateless, UI5 version of the of the game interface using the other open-source project ABAP2UI5.  I didn't want to do a pull request directly here so as not to introduce a direct, hard dependency to ABAP2UI5. 

https://github.com/jung-thomas/axage_example

What the UI looks like. Basic but workable:
![image](https://user-images.githubusercontent.com/12159356/235523065-bb7a5a37-9182-48c0-a981-c9c246f2cf3d.png)

**Comment by Ennowulff** [comment]
Hey @jung-thomas !
This is really cool! I❤️ it!
Will ASAP investigate how you did it.
Cheers
Enno

**Comment by Ennowulff** [comment]
When pulling your repo, I get the message:
`Object type HTTP is not supported by this system`

How could you export this? HTTP is not supported by abapGit!?
https://docs.abapgit.org/ref-supported.html

**Comment by jung-thomas** [comment]
I didn't expect to be in there either as I didn't think it was supported by ABAPGit. I just used the abapGit plug-in in Eclipse. The Staging view included the HTTP service and pushed it to Git. Something in the Eclipse plug-in that isn't even supported by ABAPGit itself?  

Try importing that repo again. I've cleaned everything out except the package and the ABAP2UI5 app class.

**Comment by Ennowulff** [comment]
everything is okay now, but now the main things are missing, aren't they?
![image](https://user-images.githubusercontent.com/75187288/235798074-13fa370a-3457-40af-828b-591413de4e3f.png)

```
Method Z2UI5_IF_APP~MAIN
Method "Z2UI5_IF_APP~MAIN" is not declared or inherited in class "ZCL_AXAGE_DEMO1_UI". -
Public Section ZCL_AXAGE_DEMO1_UI
Implementation missing for method "Z2UI5_IF_APP~CONTROLLER". "Z2UI5_IF_APP~CONTROLLER".

```

**Comment by jung-thomas** [comment]
No. Everything seems to be there. It should just be one class that gets imported:
![image](https://user-images.githubusercontent.com/12159356/235984791-93014803-2cd0-48b8-994e-6792cea73074.png)

I went to a fresh ABAP system and it imports from Git successfully. It also activates fine.
![image](https://user-images.githubusercontent.com/12159356/235985055-cdce9790-bf75-4849-b019-0cfaa942b60c.png)

The method ~main implementation is definitely in the class: https://github.com/jung-thomas/axage_example/blob/main/src/zcl_axage_demo1_ui.clas.abap#L24

When do you get that error?  Upon activation? If so what does your inactive version of zcl_axage_demo1_ui look like?

**Comment by Ennowulff** [comment]
Sorry, I was confused about how to use abap2ui5... 😿 

So now, when I start the abap2ui5-service and "link to the service"
![image](https://user-images.githubusercontent.com/75187288/236311874-dee0e801-4403-4687-8a33-65af2a3aee7d.png)

I get this error:
Method call failed: Method Z2UI5_IF_APP~CONTROLLER of class ZCL_AXAGE_DEMO1_UI is not implemented.

**Comment by Ennowulff** [comment]
The same old story... 😢 I haven't updated abap2ui5 to the latest release...
Now it works! Thanks again for this cool idea!
